batcher/aligned-batcher/src/lib.rs

```diff
@@ -1232,25 +1250,63 @@ impl Batcher {
                 }
                 Ok(receipt)
             }
-            Err(BatcherSendError::TransactionReverted(err)) => {
-                // Since transaction was reverted, we don't want to retry with fallback.
-                warn!("Transaction reverted {:?}", err);
-
-                Err(BatcherError::TransactionSendError)
+            Err(RetryError::Permanent(BatcherError::ReceiptNotFoundError)) => {
+                self.metrics.canceled_batches.inc();
+                self.cancel_create_new_task_tx(fee_params.gas_price).await;
+                Err(BatcherError::ReceiptNotFoundError)
             }
-            Err(_) => {
-                let receipt = try_create_new_task(
-                    batch_merkle_root,
-                    batch_data_pointer,
-                    proof_submitters,
-                    fee_params,
-                    &self.payment_service_fallback,
+            Err(_) => Err(BatcherError::TransactionSendError),
+        }
+    }
+
+    // Sends a transaction to Ethereum with the same nonce as the previous one to override it.
+    // Uses exponential backoff with the maximum number of retries because this transaction must be included.
+    // Bumps the gas price used and waits for 3 blocks on each attepmt to get the Receipt of the cancel transaction.
+    pub async fn cancel_create_new_task_tx(&self, old_tx_gas_price: U256) {
+        info!("Cancelling createNewTask transaction...");
+        let iteration = Arc::new(Mutex::new(0));
+        let previous_gas_price = Arc::new(Mutex::new(old_tx_gas_price));
+
+        if let Err(e) = retry_function(
+            || async {
+                let mut iteration = iteration.lock().await;
+                let mut previous_gas_price = previous_gas_price.lock().await;
+
+                let current_gas_price = match get_gas_price(
+                    self.batcher_signer.provider(),
+                    self.batcher_signer_fallback.provider(),
                 )
-                .await?;
+                .await
+                {
+                    Ok(gas_price) => gas_price,
+                    Err(e) => return Err(RetryError::Transient(e)),
+                };
 
-                Ok(receipt)
-            }
-        }
+                let bumped_gas_price =
+                    get_bumped_gas_price(*previous_gas_price, current_gas_price, *iteration);
```

Comment:
```suggestion
                    get_bumped_gas_price(*previous_gas_price, current_gas_price, *iteration);
```

Change name to calculate_bumped_gas_price()

Reply:
Done!